src/pkgcheck/checks/python.py

```diff
@@ -274,30 +274,34 @@ class PythonCompatCheck(Check):
 
     def __init__(self, *args):
         super().__init__(*args)
-        repo = self.options.target_repo
         # sorter for python targets leveraging USE_EXPAND flag ordering from repo
-        self.sorter = repo.use_expand_sorter('python_targets')
+        self.sorter = self.options.target_repo.use_expand_sorter('python_targets')
 
         # determine available PYTHON_TARGET use flags
-        targets = []
-        for target, _desc in repo.config.use_expand_desc.get(IUSE_PREFIX[:-1], ()):
-            if target[len(IUSE_PREFIX):].startswith('python'):
-                targets.append(target[len(IUSE_PREFIX):])
-        multi_targets = tuple(sorted(targets, key=self.sorter))
+        multi_targets = tuple(sorted(self.python_impls(IUSE_PREFIX),
+                                     key=self.sorter))
 
         # determine available PYTHON_SINGLE_TARGET use flags
-        targets = []
-        for target, _desc in repo.config.use_expand_desc.get(IUSE_PREFIX_S[:-1], ()):
-            if target[len(IUSE_PREFIX_S):].startswith('python'):
-                targets.append(target[len(IUSE_PREFIX_S):])
-        single_targets = tuple(sorted(targets, key=self.sorter))
+        single_targets = tuple(sorted(self.python_impls(IUSE_PREFIX_S),
+                                      key=self.sorter))
 
         self.params = {
             'python-r1': (multi_targets, IUSE_PREFIX, None),
             'python-single-r1': (single_targets, (IUSE_PREFIX, IUSE_PREFIX_S), None),
             'python-any-r1': (multi_targets, (IUSE_PREFIX, IUSE_PREFIX_S), ('depend', 'bdepend')),
         }
 
+    def python_impls(self, prefix):
+        # get Python implementations from profiles/desc/python_targets.desc
+        # or profiles/desc/python_single_target.desc
+        use_expand_desc = set()
+        for repo in self.options.target_repo.trees:
+            use_expand_desc.update(repo.config.use_expand_desc.get(prefix[:-1], ()))
+
+        for target, _desc in use_expand_desc:
+            if target[len(prefix):].startswith('python'):
+                yield target[len(prefix):]
+
```

Comment:
Honestly, I'd probably make this more generic and move it into pkgcore as well. Something like a function that can retrieve a given, inherited value from use_expand_desc across a target repo's dependents.